Makefile

```diff
@@ -29,14 +29,59 @@ CFLAGS = $(BASE_CFLAGS) $(BUILD_CFLAGS) $(CLFLAGS)
 LFLAGS = $(BUILD_LINK) $(LINKFLAGS)
 
 
-.cxx.obj:
-	$(CC) /c $(cflags) $(cvars) $*.cxx
+{src}.cxx{}.obj::
+	$(CC) /c $(CFLAGS) $(CVARS) /I:src $<	
 
-all: install
+{test}.cxx{test}.obj::
+	$(CC) /c $(CFLAGS) $(CVARS) /I:test $<
 
-cl.exe : cl.obj
-	link $(LFLAGS) cl.obj /out:cl.exe
+all : install
+
+cl.exe : cl.obj execute.obj intel.obj ld.obj main.obj spack_env.obj toolchain.obj toolchain_factory.obj utils.obj winrpath.obj 
+	link $(LFLAGS) $** /out:cl.exe
 
 install : cl.exe
 	mkdir $(PREFIX)
-	copy cl.exe $(PREFIX)
\ No newline at end of file
+	copy cl.exe $(PREFIX)
+	mklink /S link.exe cl.exe
+	mklink /S ifx.exe cl.exe
+	mklink /S relocate.exe cl.exe
+
+build_test_driver : test/test_driver.obj
+	rmdir /q /s tmp
+	mkdir tmp
+	lib $(LFLAGS) *.obj /out:wrapper.lib
+	link $(LFLAGS) test/test_driver.obj wrapper.lib /out:driver.exe
+	move driver.exe tmp/driver.exe
+
+setup_test: cl.exe
+	rmdir /q /s tmp
+	mkdir tmp/test
+	cd tmp/test
+	# create symlinks to test the different toolchains
+	copy ../..cl.exe cl.exe
+	mklink /S link.exe cl.exe
+	mklink /S relocate.exe cl.exe
+	cd ../..
+
+build_test_samples : test/calc.obj test/main.obj setup_test
+	link $(LFLAGS) test/calc.obj /out:calc.dll /dll
+	move calc.dll tmp/calc.dll
+	link $(LFLAGS) /test/main.obj calc.lib /out:tester.exe
+	move tester.exe tmp/test/tester.exe
```

Comment:
Is it possible to add a CI check that runs this? Does it run now? 

I'm assuming not given

* issue w/`setup_test`
* how is the `link` command above able to know that `calc.dll` is in `tmp` in order to generate the abspath?

Reply:
tester.exe runs if setup locally (roughly following these instructions), this makefile test does not work as of yet.